algorithms/estonian/stemmer.tt

```diff
@@ -0,0 +1,479 @@
+        <DD>replace by <B><I>võisi</I></B>
+    <DT><B><I>jään  &nbsp;  jääd  &nbsp;  jääb  &nbsp;  jääme  &nbsp;  jääte  &nbsp;  jäävad</I></B>
+        <DD>replace by <B><I>jääma</I></B>
+    <DT><B><I>jääksin  &nbsp;  jääksid  &nbsp;  jääks  &nbsp;  jääksime  &nbsp;  jääksite</I></B>
+        <DD>replace by <B><I>jääma</I></B>
+    <DT><B><I>jäime  &nbsp;  jäite  &nbsp;  jäin  &nbsp;  jäid  &nbsp;  jäi</I></B>
+        <DD>replace by <B><I>jääma</I></B>
+    <DT><B><I>müün  &nbsp;  müüd  &nbsp;  müüb  &nbsp;  müüs  &nbsp;  müüme  &nbsp;  müüte  &nbsp;  müüvad</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>müüksin  &nbsp;  müüksid  &nbsp;  müüks  &nbsp;  müüksime  &nbsp;  müüksite</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>müüakse  &nbsp;  müüdi  &nbsp;  müüa  &nbsp;  müüma  &nbsp;  müümata</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>loeb  &nbsp;  loen  &nbsp;  loed  &nbsp;  loeme  &nbsp;  loete  &nbsp;  loevad</I></B>
+        <DD>replace by <B><I>luge</I></B>
+    <DT><B><I>loeks  &nbsp;  loeksin  &nbsp;  loeksid  &nbsp;  loeksime  &nbsp;  loeksite</I></B>
+        <DD>replace by <B><I>luge</I></B>
+    <DT><B><I>põen  &nbsp;  põeb  &nbsp;  põed  &nbsp;  põeme  &nbsp;  põete &nbsp;  põevad</I></B>
+        <DD>replace by <B><I>põde</I></B>
+    <DT><B><I>põeksin  &nbsp;  põeks  &nbsp;  põeksid  &nbsp;  põeksime  &nbsp;  põeksite</I></B>
+        <DD>replace by <B><I>põde</I></B>
+    <DT><B><I>laon  &nbsp;  laob  &nbsp;  laod  &nbsp;  laome  &nbsp;  laote  &nbsp;  laovad</I></B>
+        <DD>replace by <B><I>ladu</I></B>
+    <DT><B><I>laoksin  &nbsp;  laoks  &nbsp;  laoksid  &nbsp;  laoksime  &nbsp;  laoksite</I></B>
+        <DD>replace by <B><I>ladu</I></B>
+    <DT><B><I>teeksin  &nbsp;  teeks  &nbsp;  teeksid  &nbsp;  teeksime  &nbsp;  teeksite</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>teen  &nbsp;  teeb  &nbsp;  teed  &nbsp;  teeme  &nbsp;  teete  &nbsp;  teevad</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>tehakse  &nbsp;  tehti  &nbsp;  tegema  &nbsp;  tegemata  &nbsp;  teha</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>näen  &nbsp;  näeb  &nbsp;  näed  &nbsp;  näeme  &nbsp;  näete  &nbsp;  näevad</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+    <DT><B><I>näeksin  &nbsp;  näeks  &nbsp;  näeksid  &nbsp;  näeksime  &nbsp;  näeksite</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+    <DT><B><I>nähakse  &nbsp;  nähti  &nbsp;  näha  &nbsp;  nägema  &nbsp;  nägemata</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+</DL>
+</DL>
+
+
+<p>
+Step 1: emphasis
+</p>
+
+<DL><DD>
+    Search for the longest among the following suffixes in <I>R1</I>, and perform the action indicated
+<BR><BR>
+<DL>
+        <DD>Test if there’s at least 4 letters before <I>R1</I> region. If so, continue this step
+    <DT><B><I>gi</I></B>
+        <DD>if preceded by <B><I>GI</I></B> and <B>not <I>LONGV</I></B>, delete
```

Comment:
I think this would be clearer phrased as something like "if preceded by a character from _GI_ which isn't the second character of a long vowel as defined by _LONGV_" - otherwise it assumes the reader understands how `and` works in Snowball, and that this "and" is meant to be interpreted like Snowball's `and`.

Reply:
I've made this change (I did consider whether the intention was "if preceded by **GI** which is not preceded by **LONGV**", but the motivating example from `estonian.sbl` of "bioloogi -> bioloogi" wouldn't work if that was the intent).